--- git status ---
On branch master
Your branch is ahead of 'origin/master' by 1 commit.
  (use "git push" to publish your local commits)

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   scripts/rsl_rl/__pycache__/cli_args.cpython-311.pyc
	modified:   source/rob6323_go2/rob6323_go2/tasks/direct/rob6323_go2/rob6323_go2_env.py
	modified:   source/rob6323_go2/rob6323_go2/tasks/direct/rob6323_go2/rob6323_go2_env_cfg.py

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/scripts/rsl_rl/__pycache__/cli_args.cpython-311.pyc b/scripts/rsl_rl/__pycache__/cli_args.cpython-311.pyc
index cfe10a7..f2cdd8a 100644
Binary files a/scripts/rsl_rl/__pycache__/cli_args.cpython-311.pyc and b/scripts/rsl_rl/__pycache__/cli_args.cpython-311.pyc differ
diff --git a/source/rob6323_go2/rob6323_go2/tasks/direct/rob6323_go2/rob6323_go2_env.py b/source/rob6323_go2/rob6323_go2/tasks/direct/rob6323_go2/rob6323_go2_env.py
index 031975f..2b28417 100644
--- a/source/rob6323_go2/rob6323_go2/tasks/direct/rob6323_go2/rob6323_go2_env.py
+++ b/source/rob6323_go2/rob6323_go2/tasks/direct/rob6323_go2/rob6323_go2_env.py
@@ -189,7 +189,7 @@ class Rob6323Go2Env(DirectRLEnv):
         # === ADDED: Feet clearance penalty ===
         phases = 1 - torch.abs(1.0 - torch.clip((self.foot_indices * 2.0) - 1.0, 0.0, 1.0) * 2.0)
         foot_heights = self.foot_positions_w[:, :, 2]
-        target_height = 0.08 * phases + 0.02
+        target_height = 0.1 * phases + 0.02
         rew_foot_clearance = torch.square(target_height - foot_heights) * (1 - self.desired_contact_states)
         rew_feet_clearance = torch.sum(rew_foot_clearance, dim=1)
 
diff --git a/source/rob6323_go2/rob6323_go2/tasks/direct/rob6323_go2/rob6323_go2_env_cfg.py b/source/rob6323_go2/rob6323_go2/tasks/direct/rob6323_go2/rob6323_go2_env_cfg.py
index b5f008d..1351b00 100644
--- a/source/rob6323_go2/rob6323_go2/tasks/direct/rob6323_go2/rob6323_go2_env_cfg.py
+++ b/source/rob6323_go2/rob6323_go2/tasks/direct/rob6323_go2/rob6323_go2_env_cfg.py
@@ -36,7 +36,7 @@ class Rob6323Go2EnvCfg(DirectRLEnvCfg):
     state_space = 0
     debug_vis = True
     # === ADDED: Base height termination threshold ===
-    base_height_min = 0.1
+    base_height_min = 0.05
 
     # simulation
     sim: SimulationCfg = SimulationCfg(
@@ -70,7 +70,7 @@ class Rob6323Go2EnvCfg(DirectRLEnvCfg):
         prim_path="/World/ground",
         terrain_type="generator",
         terrain_generator=TerrainGeneratorCfg(
-            size=(100.0, 100.0), #(8.0, 8.0)
+            size=(200.0, 200.0), #(8.0, 8.0)
             border_width=0.0, #20.0
             num_rows=1, #10
             num_cols=1, #20
@@ -81,7 +81,7 @@ class Rob6323Go2EnvCfg(DirectRLEnvCfg):
             sub_terrains={
                 "random_rough": terrain_gen.HfRandomUniformTerrainCfg(
                     proportion=1.0,  # 100% random rough terrain (continuous, simple)
-                    noise_range=(0.01, 0.07),  # Low noise range (1-4cm) for simple terrain
+                    noise_range=(0.01, 0.07), 
                     noise_step=0.01,
                     border_width=0.25,
                 ),
@@ -150,7 +150,7 @@ class Rob6323Go2EnvCfg(DirectRLEnvCfg):
     # === ADDED: Angular velocity penalty (roll/pitch) ===
     ang_vel_xy_reward_scale = -0.001
     # === ADDED: Feet clearance penalty ===
-    feet_clearance_reward_scale = -30.0 # -30.0
+    feet_clearance_reward_scale = -200.0 # (was -30.0) Notes: Need to increase this to force the robot to lift its feet more often and "climb" more easily
     # === ADDED: Contact tracking shaped force ===
     tracking_contacts_shaped_force_reward_scale = 0.0
 